service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/authentication/ServiceClientCredentialsProvider.java

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.azure.sdk.iot.service.digitaltwin.authentication;;
+
+import com.microsoft.rest.credentials.ServiceClientCredentials;
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+
+@AllArgsConstructor
+public class ServiceClientCredentialsProvider implements ServiceClientCredentials {
+
+    private static final String AUTHORIZATION = "Authorization";
+    @NonNull
+    private final SasTokenProvider sasTokenProvider;
+
+    @Override
+    public void applyCredentialsFilter(OkHttpClient.Builder clientBuilder) {
+        Interceptor authenticationInterceptor = chain -> {
+            Request authenticatedRequest = chain.request()
+                                                .newBuilder()
```

Comment:
nit : indentation

Reply:
This one and two below, I have left as it is. Wasn't sure if you meant it is more indented but looked fine to me.